File: Java/DesignPatterns/src/test/java/designPatterns/behavioral/interpreter/MainInterpreterTest.java
package designPatterns.behavioral.interpreter;

import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertTrue;

public class MainInterpreterTest {
  private Expression isMale;
  private Expression isMarried;
  private Expression isMarriedAndMale;

  @BeforeEach
  void setUp() {
    isMale = new ConcreteExpression.TerminalExpression("John");
    isMarried = new ConcreteExpression.TerminalExpression("Married");
    isMarriedAndMale = new ConcreteExpression.AndExpression(isMale, isMarried);
  }

  @Test
  void equalsTest() {
    assertTrue(isMale.interpret("John"));
    assertTrue(isMarried.interpret("Married"));
    assertTrue(isMarriedAndMale.interpret("John Married"));
  }
}

File: Kotlin/DesignPatterns/src/test/kotlin/behavioral/interpreter/MainInterpreterTest.kt
package behavioral.interpreter

import org.junit.jupiter.api.BeforeEach
import org.junit.jupiter.api.Test
import kotlin.test.assertTrue

class MainInterpreterTest {
  private lateinit var isMale: Expression
  private lateinit var isMarried: Expression
  private lateinit var isMarriedAndMale: Expression

  @BeforeEach
  internal fun setUp() {
    isMale = ConcreteExpression.TerminalExpression("John")
    isMarried = ConcreteExpression.TerminalExpression("Married")
    isMarriedAndMale = ConcreteExpression.AndExpression(isMale, isMarried)
  }

  @Test
  internal fun equalsTest() {
    assertTrue(isMale.interpret("John"))
    assertTrue(isMarried.interpret("Married"))
    assertTrue(isMarriedAndMale.interpret("John Married"))
  }
}